taichi/transforms/check_out_of_bound.cpp

```diff
@@ -104,10 +111,37 @@ class CheckOutOfBound : public BasicStmtVisitor {
     set_done(stmt);
   }
 
+  void visit(BinaryOpStmt *stmt) override {
+    // Insert assertions if debug is on
+    if (is_done(stmt)) {
+      return;
+    }
+    if (stmt->op_type == BinaryOpType::pow) {
+      if (is_integral(stmt->rhs->ret_type) &&
+          is_integral(stmt->lhs->ret_type)) {
+        if (config_.debug) {
```

Comment:
This line may not be necessary.
FYI
https://github.com/taichi-dev/taichi/blob/d4f4abf289f75ec17663c0dbf2ef690112580e4e/taichi/program/program.cpp#L76-L77
https://github.com/taichi-dev/taichi/blob/d4f4abf289f75ec17663c0dbf2ef690112580e4e/taichi/transforms/compile_to_offloads.cpp#L99-L103